tests/functional/d/deprecated/dataclass_typecheck.txt

```diff
@@ -0,0 +1,12 @@
"""Return value^self.""""""\n        \n    \n    def __setattr__():\n        """"""Implement setattr(self, name, value).""""""\n        \n    \n    def __sizeof__():\n        """"""Returns size in memory, in bytes.""""""\n        \n    \n    def __str__():\n        """"""Return str(self).""""""\n        \n    \n    def __sub__():\n        """"""Return self-value.""""""\n        \n    \n    def __subclasshook__():\n        """"""Abstract classes can override this to customize issubclass().\n\nThis is invoked early on by abc.ABCMeta.__subclasscheck__().\nIt should return True, False or NotImplemented.  If it returns\nNotImplemented, the normal algorithm is used.  Otherwise, it\noverrides the normal algorithm (and the outcome is cached).\n""""""\n        \n    \n    def __truediv__():\n        """"""Return self/value.""""""\n        \n    \n    def __trunc__():\n        """"""Truncating an Integral returns itself.""""""\n        \n    \n    def __xor__():\n        """"""Return self^value.""""""\n        \n    \n    def as_integer_ratio():\n        """"""Return integer ratio.\n\nReturn a pair of integers, whose ratio is exactly equal to the original int\nand with a positive denominator.\n\n>>> (10).as_integer_ratio()\n(10, 1)\n>>> (-10).as_integer_ratio()\n(-10, 1)\n>>> (0).as_integer_ratio()\n(0, 1)""""""\n        \n    \n    def bit_length():\n        """"""Number of bits necessary to represent self in binary.\n\n>>> bin(37)\n\'0b100101\'\n>>> (37).bit_length()\n6""""""\n        \n    \n    def conjugate():\n        """"""Returns self, the complex conjugate of any int.""""""\n        \n    \n    \n    class denominator:\n        """"""the denominator of a rational number in lowest terms""""""\n        \'denominator\'\n        \'denominator\'\n    \n    \n    \n    class imag:\n        """"""the imaginary part of a complex number""""""\n        \'imag\'\n        \'imag\'\n    \n    \n    \n    class numerator:\n        """"""the numerator of a rational number in lowest terms""""""\n        \'numerator\'\n        \'numerator\'\n    \n    \n    \n    class real:\n        """"""the real part of a complex number""""""\n        \'real\'\n        \'real\'\n    \n    \n    def to_bytes():\n        """"""Return an array of bytes representing an integer.\n\n  length\n    Length of bytes object to use.  An OverflowError is raised if the\n    integer is not representable with the given number of bytes.\n  byteorder\n    The byte order used to represent the integer.  If byteorder is \'big\',\n    the most significant byte is at the beginning of the byte array.  If\n    byteorder is \'little\', the most significant byte is at the end of the\n    byte array.  To request the native byte order of the host system, use\n    `sys.byteorder\' as the byte order value.\n  signed\n    Determines whether two\'s complement is used to represent the integer.\n    If signed is False and a negative integer is given, an OverflowError\n    is raised.""""""\n        \n' used":HIGH
```

Comment:
Wow. Do you know how we could we fix that ? I know this is not really the goal of this MR but this is... bad.

Reply:
I'm guessing an `as_string` is being called on a node, but it's not immediately clear to me why. I can look into this in a bit, let me know if you want me to add to this PR or open up a new one.

Reply:
@Pierre-Sassoulas okay never mind, I fixed the immediate problem in this MR. I think the `builtins.int` class that astroid defines has different methods depending on Python version, which caused the test output to be inconsistent.

Reply:
:+1: I think it's a mistake, there's no way we have such a messy warning on purpose :smile: Could you fix this in another MR then rebase this one on main when we've merged it ? :)